### PR TITLE
Implement ElementHandle.ScreenshotStreamAsync

### LINKED ISSUE #30: Implement ElementHandle.Screenshot tests
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L1883

### PR DESCRIPTION
**Pull request by kblok**
closes #30

### CODE REVIEW

**Comment by kblok** [comment]
Ready to review @Meir017

File: lib/PuppeteerSharp/Page.cs [hunk 1]
```diff
@@ -677,8 +699,17 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
             }
         }
 
+        /// <summary>
+        /// Takes a screenshot of the page
+        /// </summary>
+        /// <returns>The screenshot task returning the image stream.</returns>
         public async Task<Stream> ScreenshotStreamAsync() => await ScreenshotStreamAsync(new ScreenshotOptions());
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: Redundant async await

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -656,12 +663,27 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
...
             SetUserAgentAsync(options.UserAgent)
         );
 
+        /// <summary>
+        /// Takes a screenshot of the page
+        /// </summary>
+        /// <returns>The screenshot task.</returns>
+        /// <param name="file">The file path to save the image to. The screenshot type will be inferred from file extension. 
+        /// If path is a relative path, then it is resolved relative to current working directory. If no path is provided, 
+        /// the image won't be saved to the disk.</param>
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: Redundant async await

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -656,12 +663,27 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
         }
 
         public async Task EmulateAsync(DeviceDescriptor options) => await Task.WhenAll(
-            SetViewport(options.ViewPort),
+            SetViewportAsync(options.ViewPort),
             SetUserAgentAsync(options.UserAgent)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: Redundant async await

**Comment by kblok** [comment]
Done @Meir017

**Comment by Meir017** [comment]
CodeFactor found a duplicated code 😏

File: lib/PuppeteerSharp/Page.cs [hunk 4]
```diff
@@ -644,7 +644,14 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
...
 
-        public async Task SetViewport(ViewPortOptions viewport)
+        /// <summary>
+        /// Sets the viewport.
+        /// In the case of multiple pages in a single browser, each page can have its own viewport size.
+        /// NOTE in certain cases, setting viewport will reload the page in order to set the isMobile or hasTouch properties.
+        /// </summary>
+        /// <returns>The viewport task.</returns>
+        /// <param name="viewport">Viewport options.</param>
+        public async Task SetViewportAsync(ViewPortOptions viewport)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: maybe rename to `SetViewPortAsync`?

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: The method is `setViewport` on Puppeteer.

**Comment by kblok** [comment]
Regarding the CodeFactor issue @Meir017. I don't buy that. We won't create a base class just to reuse those Screenshots overrides.

**Comment by Meir017** [comment]
How about a static internal method on one the yhe classes?

**Comment by kblok** [comment]
@Meir017  The thing is that those 3 "duplicated" methods end up calling its own ` public async Task<Stream> ScreenshotStreamAsync(ScreenshotOptions options)` and the implementation is different there. 

Or am I missing something?

**Comment by Meir017** [comment]
Ok

**Pull request merged**